core/unix/os_private.h

```diff
@@ -269,6 +269,55 @@ os_walk_address_space(memquery_iter_t *iter, bool add_modules);
 bool
 is_sigreturn_syscall_number(int sysnum);
 
+struct os_glob_t {
+    size_t gl_pathc;
+    char **gl_pathv;
+    size_t gl_offs;
+};
+
+typedef struct os_glob_t os_glob_t;
+
+#ifdef GLOB_MARK
+#    undef GLOB_MARK
+#endif
+#define GLOB_MARK (1 << 0)
+
+#ifdef GLOB_DOOFFS
```

Comment:
Wondering whether instead of an identical interface to libc we should provide just what we need, which is a narrower interface and thus easier to test and maintain.  We only need to glob directories so we could not take any options and only provide a directory-globbing routine.  If in the future we needed further glob functionality we could at that point switch back to something like this (since it's all internal and not part of DR's exposed API).  WDYT?

Reply:
I think this makes sense since `glob` has a number of options that we do not need and that are not trivial to implement like not setting `GLOB_NOSORT`, in which case we need to call `qsort` (DynamoRIO might already have this and I simply missed it).

In addition, I want to keep the number of libc functions that we have to implement to a minimum, since some of them can be hard to get right and both `fnmatch` and `glob` are already non-trivial but unfortunately necessary to deal with `/etc/ld.so.conf`.